src/auth/authenticator/tls/authenticator.go

```diff
@@ -0,0 +1,61 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/x509"
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/authenticator"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+	"net/http"
+)
+
+const AuthMode = "Tls"
+
+type Authenticator struct {
+	Certificates []*x509.Certificate
+}
+
+var _ authenticator.Authenticator = Authenticator{}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{Certificates: cfg.Certificates}
+
+	return auth, nil
+}
+
+// Authenticate checks certs and retrieve user Info
+// CN field -> Username
+// Organisation -> Groups
+func (authenticator Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// extract Certificate info from request if any
+	if len(req.TLS.PeerCertificates) > 0 {
+		// As mentioned in doc first array element is the leaf
```

Comment:
This annotation won't mean much in few weeks as we won't know which doc we refer to. IMO it would be better if we look for the actual tecnical reason making this statement true

Reply:
This one is the PeerCertificate doc, should not move too much I think. Anyway mentioning the doc here is not clear. Ok for me